packages/salesforcedx-vscode-apex-debugger/src/core/sessionService.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  CommandExecution,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { CommandOutput } from '../utils/commandOutput';
+
+export class SessionService {
+  private static instance: SessionService;
+  private userFilter: string;
+  private requestFilter: string;
+  private entryFilter: string;
+  private project: string;
+  private sessionId: string;
+  private connected: boolean;
+
+  public static getInstance() {
+    if (!SessionService.instance) {
+      SessionService.instance = new SessionService();
+    }
+    return SessionService.instance;
+  }
+
+  public withUserFilter(filter: string): SessionService {
+    this.userFilter = filter || '';
```

Comment:
That default value is for auto-completion. The user can omit userIdFilter in their config, and we would get the property as 'undefined'.